src/parser.js

```diff
@@ -1157,12 +1158,19 @@ export default class Parser {
       Type = BooleanElement;
     } else if (parameter.type === 'array') {
       Type = ArrayElement;
+
+      if (example && !Array.isArray(example)) {
```

Comment:
The check is only for array types, I think we could make this generic and support all types. What do you think?

Mostly (possibly working) pseudo code but you get the idea:

```js
const parameterTypes = {
  string: {
    element: StringElement,
    type: String
  },
  array: {
    element: ArrayElement,
    type: Array
  },
  // all types go here
}

const parameterType = parameterTypes[parameter.type];
if (parameterType) {
  const element = new parameterType.element();

  if (example) {
    if (example instance parameterType.type) {
      element.content = example;
      // todo set source map?
    } else {
      // create annotation of example being wrong type
    }
  }
} else {
  // create annotation of unsupported/unknown type
}
```